WordPress/create-block-theme
# Editor Sidebar: Persist "Save Changes" panel settings

Closes #597

This PR persists the checked state of the "Save Changes" panel.

To make settings persistent, use the [@wordpress/preferences](https://github.com/WordPress/gutenberg/blob/trunk/packages/preferences/README.md) package. This package is also used in Gutenberg to persist various settings, which are stored in a database. This ensures that the same settings remain the same for the same user even if the user reloads the browser or uses a different browser.

![database-info](https://github.com/WordPress/create-block-theme/assets/54422211/f63ff27d-3304-4f7b-b052-c70b66b0fc6b)

### Testing Instructions

- By default, the following four items should be checked.
  - Save Fonts
  - Save Style Changes
  - Save Template Changes
  - Process Only Modified Templates
- When the "Save template Changes" is unchecked, the four checkboxes below it will be disabled as before.
- Reload your browser and confirm that the settings persist.
- Confirm that when you click "Save Changes" the theme will update according to your settings as before.

### About Initializing Settings

To restart the test from the beginning, perform the following two operations.

Chrome Dev Tools > Apprication > Storage > Local storage > Right click "http://localhost:8888" > Clear

![local-storage](https://github.com/WordPress/create-block-theme/assets/54422211/e328910b-03da-4f8a-8689-cb0a8c4416a7)

Connect to database > `wp_usermeta` table > Delete `wp_persisted_preferences` record

![database](https://github.com/WordPress/create-block-theme/assets/54422211/c97d7f22-4836-41ac-bc7a-360d9fa75120)
